### PR TITLE
Implement gene-list exporter

### PR DESCRIPTION
**Pull request by dmgaldi**
Ported from [EuPathGalaxy](https://github.com/VEuPathDB/EuPathGalaxy). Tested with updates to the utility server https://github.com/VEuPathDB/util-user-dataset-handler-server/pull/24 and verified files made it to IRODS.

There are still some issues with the IRODS installer but I'm not able to [re-enable the job](https://ws.qa.apidb.org/job/irods-handler-PlasmoDB/)

### CODE REVIEW

File: lib/python/eupath/GeneListExporter.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if len(args) < 8:
+            raise ValidationException("The tool was passed an insufficient numbers of arguments.")
+
+        # Data for the input given by the user
+        self._dataset_file_path = args[7]
+
+        # Overriding the dataset genome reference with that provided via the form.
+        if len(args[6].strip()) == 0:
+            raise ValidationException("ProjectIds must be specified.")
+        self._projects = args[6].split(",")
```
Comment (dmgaldi) [inline lib/python/eupath/GeneListExporter.py]: This is pretty much the only change from the Galaxy code

File: lib/python/eupath/GeneListExporter.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                                GeneListExport.GENE_LIST_VERSION,
+                                GeneListExport.GENE_LIST_VALIDATION_SCRIPT,
+                                args)
+
+        # For the gene list export, three parameters beyond generic 7 are required.
+        if len(args) < 8:
+            raise ValidationException("The tool was passed an insufficient numbers of arguments.")
+
+        # Data for the input given by the user
+        self._dataset_file_path = args[7]
```
Comment (ryanrdoherty) [inline lib/python/eupath/GeneListExporter.py]: In Galaxy, I think there may be some guarantees about the format of the file being exported.  If the user is sending us a gene list from a step, then that is also handled (albeit client-side), but the user could literally upload any file they want here- wondering if/where we are validating at least that this is a text file with a column of words?  If we're not, we probably should.

**Review by ryanrdoherty — APPROVED** [review]
General comment- not sure using "exporter" in the name here is appropriate as it was in Galaxy.  Maybe replace in the python class and script with "handler"?  Also, can you fix up the README to reflect what this specific handler does vs. the info about the example?

File: lib/python/eupath/GeneListExporter.py (hunk 2 above)
Comment (Foxcapades) [reply lib/python/eupath/GeneListExporter.py]: The only validation that happens automatically before this script is called is that the file extension aligns with allowed values.  So any additional validation (such as confirming TSV formatting) will probably have to be added to the python code.

Comment (dmgaldi) [reply lib/python/eupath/GeneListExporter.py]: Got it, I made some README updates, I'll update this PR with some validation as well and possibly README updates as well

**Pull request merged**